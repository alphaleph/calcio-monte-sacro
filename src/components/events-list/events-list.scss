@import '../../pages/globals.scss';

.is-most-recent {
    border: 4px solid $roma-tyrian;
}

.most-recent-note {
    background: $beige-main;
    text-align: right;
    font-weight: bold;
    font-size: 1.25rem;
    padding: 1rem;
}