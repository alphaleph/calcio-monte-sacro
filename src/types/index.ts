import React from 'react';

export interface PageTitleToURLMap {
    [pageTitle: string]: string;
}

export interface TypeToElementMap {
    [modalType: string]: React.ReactNode;
}
