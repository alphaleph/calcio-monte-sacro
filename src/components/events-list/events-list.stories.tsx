import React from 'react';
import { storiesOf } from '@storybook/react';
import { PureEventsList } from './events-list';
import '../../pages/index.scss';
import { mockEventsDataNodes } from '../__tests__/test-util';

storiesOf('EventsList', module)
    .add('default', () => <PureEventsList events={mockEventsDataNodes} />)
    .add('empty', () => <PureEventsList events={[]} />);
