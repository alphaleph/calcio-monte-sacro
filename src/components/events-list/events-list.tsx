import React from 'react';
import { graphql, useStaticQuery } from 'gatsby';
import { EventRow } from '../event-row/event-row';
import { MarkdownData, MarkdownEventDataNode } from '../../types/index';
import { findMostRecentlyPastIndex } from '../../util';
import './events-list.scss';

interface EventsListOwnProps {
    date?: number;
    isDesc?: boolean;
}

interface EventsListProps extends EventsListOwnProps {
    events: MarkdownEventDataNode[];
}

export const PureEventsList = ({
    events,
    date = Date.now(),
    isDesc = true,
}: EventsListProps) => {
    if (events.length > 0) {
        const mostRecentlyPastIndex = findMostRecentlyPastIndex(events, date);

        const ascRowList = events.map((event, index) =>
            index === mostRecentlyPastIndex ? (
                <li className="is-most-recent" key={index}>
                    <p className="most-recent-note">Latest Event</p>
                    <EventRow event={event ? event : null} />
                </li>
            ) : (
                <li className="" key={index}>
                    <EventRow event={event ? event : null} />
                </li>
            )
        );

        const rowList = isDesc ? [...ascRowList].reverse() : ascRowList;

        return (
            <ul className="events-list container" data-testid="events-list">
                {rowList}
            </ul>
        );
    } else {
        return (
            <section
                className="section has-background-team-granite has-text-team-holy has-text-centered has-text-weight-bold is-size-4"
                data-testid="events-list-empty"
            >
                There are currently no events!
                <br />
                <span className="is-size-1" role="img" aria-label="zzz">
                    ðŸ’¤
                </span>
            </section>
        );
    }
};

export const EventsList = (props: EventsListOwnProps) => {
    const data: MarkdownData = useStaticQuery(
        graphql`
            query {
                allMarkdownRemark(
                    sort: { order: ASC, fields: frontmatter___date }
                    filter: { frontmatter: { layout: { eq: "schedule" } } }
                ) {
                    edges {
                        node {
                            frontmatter {
                                title
                                language
                                type
                                venue
                                opposing_team
                                score_ms
                                score_other
                                date
                                layout
                            }
                            fields {
                                slug
                            }
                            html
                        }
                    }
                }
            }
        `
    );

    return (
        <PureEventsList
            {...props}
            events={data.allMarkdownRemark.edges as MarkdownEventDataNode[]}
        />
    );
};
