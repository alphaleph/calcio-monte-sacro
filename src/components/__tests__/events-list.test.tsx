import React from 'react';
import { PureEventsList } from '../events-list/events-list';
import { mockEventsDataNodes } from '../__tests__/test-util';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MarkdownEventDataNode } from '../../types';

const EMPTY_TEST_ID = 'events-list-empty';
const LIST_TEST_ID = 'events-list';
const EMPTY_LIST: MarkdownEventDataNode[] = [];

//TODO: Props Tests: Date, isDesc

describe('Events List', () => {
    it('renders without crashing', () => {
        const { getByTestId } = render(
            <PureEventsList events={mockEventsDataNodes} />
        );
        expect(getByTestId(LIST_TEST_ID)).toBeInTheDocument();
    });

    it('renders without crashing--empty', () => {
        const { getByTestId } = render(<PureEventsList events={EMPTY_LIST} />);
        expect(getByTestId(EMPTY_TEST_ID)).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const { getByTestId } = render(
            <PureEventsList events={mockEventsDataNodes} />
        );
        expect(getByTestId(LIST_TEST_ID)).toBeVisible();
    });

    it('renders correctly--empty', () => {
        const { getByTestId } = render(<PureEventsList events={EMPTY_LIST} />);
        expect(getByTestId(EMPTY_TEST_ID)).toBeVisible();
    });

    it('has not changed', () => {
        const { asFragment } = render(
            <PureEventsList events={mockEventsDataNodes} />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('has not changed--empty', () => {
        const { asFragment } = render(<PureEventsList events={EMPTY_LIST} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
