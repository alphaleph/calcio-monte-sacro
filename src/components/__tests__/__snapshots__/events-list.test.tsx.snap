// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events List has not changed 1`] = `
<DocumentFragment>
  <ul
    class="container"
    data-testid="events-list"
  >
    <li
      class=""
    >
      <article
        class="event-row-container container"
      >
        <header
          class="event-row-header"
        >
          <p>
            20 NOV 2200
          </p>
          <p
            class="has-text-weight-light"
          >
            Parco Sannazzaro
          </p>
        </header>
        <section
          class="event-row-info columns is-mobile is-vcentered"
        >
          <div
            class="column is-two-fifths has-text-right team-info"
          >
            <span
              class="team-info-abrv"
            >
              CMS
            </span>
            <span
              aria-label="Soccer ball"
              class="team-info-icon"
              role="img"
            >
              âš½
            </span>
          </div>
          <div
            class="column is-one-fifth match-info has-text-roma-red has-text-centered"
          >
            <p>
              12:40
            </p>
          </div>
          <div
            class="column is-two-fifths team-info"
          >
            <span
              aria-label="Soccer ball"
              class="team-info-icon"
              role="img"
            >
              âš½
            </span>
            <span
              class="team-info-abrv"
            >
              CMS2
            </span>
          </div>
        </section>
      </article>
    </li>
    <li
      class=""
    >
      <article
        class="event-row-container container"
      >
        <header
          class="event-row-header"
        >
          <p>
            20 NOV 2200
          </p>
          <p
            class="has-text-weight-light"
          >
            Parco Sannazzaro
          </p>
        </header>
        <section
          class="event-row-info columns is-mobile is-vcentered"
        >
          <div
            class="column is-two-fifths has-text-right team-info"
          >
            <span
              class="team-info-abrv"
            >
              CMS
            </span>
            <span
              aria-label="Soccer ball"
              class="team-info-icon"
              role="img"
            >
              âš½
            </span>
          </div>
          <div
            class="column is-one-fifth match-info has-text-roma-red has-text-centered"
          >
            <p>
              12:40
            </p>
          </div>
          <div
            class="column is-two-fifths team-info"
          >
            <span
              aria-label="Soccer ball"
              class="team-info-icon"
              role="img"
            >
              âš½
            </span>
            <span
              class="team-info-abrv"
            >
              CMS2
            </span>
          </div>
        </section>
      </article>
    </li>
    <li
      class="is-most-recent"
    >
      <p
        class="most-recent-note"
      >
        Latest Event
      </p>
      <article
        class="event-row-container container"
      >
        <header
          class="event-row-header"
        >
          <p>
            20 NOV 2019
          </p>
          <p
            class="has-text-weight-light"
          >
            Parco Sannazzaro
          </p>
        </header>
        <section
          class="event-row-info columns is-mobile is-vcentered"
        >
          <div
            class="column is-two-fifths has-text-right team-info"
          >
            <span
              class="team-info-abrv"
            >
              CMS
            </span>
            <span
              aria-label="Soccer ball"
              class="team-info-icon"
              role="img"
            >
              âš½
            </span>
          </div>
          <div
            class="column is-one-fifth match-info has-text-roma-red has-text-centered"
          >
            <p>
              <span
                class="match-info-score"
              >
                1
              </span>
              <span
                class="has-text-team-granite"
              >
                -
              </span>
              <span
                class="match-info-score"
              >
                2
              </span>
            </p>
          </div>
          <div
            class="column is-two-fifths team-info"
          >
            <span
              aria-label="Soccer ball"
              class="team-info-icon"
              role="img"
            >
              âš½
            </span>
            <span
              class="team-info-abrv"
            >
              CMS2
            </span>
          </div>
        </section>
      </article>
    </li>
    <li
      class=""
    >
      <article
        class="event-row-container container"
      >
        <header
          class="event-row-header"
        >
          <p>
            20 NOV 2019
          </p>
          <p
            class="has-text-weight-light"
          >
            Parco Sannazzaro
          </p>
        </header>
        <section
          class="event-row-info columns is-mobile is-vcentered"
        >
          <div
            class="column is-two-fifths has-text-right team-info"
          >
            <span
              class="team-info-abrv"
            >
              CMS
            </span>
            <span
              aria-label="Soccer ball"
              class="team-info-icon"
              role="img"
            >
              âš½
            </span>
          </div>
          <div
            class="column is-one-fifth match-info has-text-roma-red has-text-centered"
          >
            <p>
              <span
                class="match-info-score"
              >
                1
              </span>
              <span
                class="has-text-team-granite"
              >
                -
              </span>
              <span
                class="match-info-score"
              >
                2
              </span>
            </p>
          </div>
          <div
            class="column is-two-fifths team-info"
          >
            <span
              aria-label="Soccer ball"
              class="team-info-icon"
              role="img"
            >
              âš½
            </span>
            <span
              class="team-info-abrv"
            >
              CMS2
            </span>
          </div>
        </section>
      </article>
    </li>
    <li
      class=""
    >
      <article
        class="event-row-container container"
      >
        <header
          class="event-row-header"
        >
          <p>
            20 NOV 2019
          </p>
          <p
            class="has-text-weight-light"
          >
            Parco Sannazzaro
          </p>
        </header>
        <section
          class="event-row-info has-text-centered columns is-mobile is-vcentered"
        >
          <div
            class="column practice-info"
          >
            <p>
              PRACTICE
            </p>
            <p
              class="has-text-roma-red"
            >
              12:40
            </p>
          </div>
        </section>
      </article>
    </li>
    <li
      class=""
    >
      <article
        class="event-row-container container"
      >
        <header
          class="event-row-header"
        >
          <p>
            20 NOV 2019
          </p>
          <p
            class="has-text-weight-light"
          >
            Parco Sannazzaro
          </p>
        </header>
        <section
          class="event-row-info has-text-centered columns is-mobile is-vcentered"
        >
          <div
            class="column practice-info"
          >
            <p>
              PRACTICE
            </p>
            <p
              class="has-text-roma-red"
            >
              12:40
            </p>
          </div>
        </section>
      </article>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`Events List has not changed--empty 1`] = `
<DocumentFragment>
  <section
    class="section has-background-team-granite has-text-team-holy has-text-centered  has-text-weight-bold is-size-4"
    data-testid="events-list-empty"
  >
    There are currently no events!
    <br />
    <span
      aria-label="zzz"
      class="is-size-1"
      role="img"
    >
      ðŸ’¤
    </span>
  </section>
</DocumentFragment>
`;
